src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -483,7 +495,32 @@ public Builder superclass(TypeName superclass) {
     }
 
     public Builder superclass(Type superclass) {
-      return superclass(TypeName.get(superclass));
+      return superclass(superclass, true);
+    }
+
+    public Builder superclass(Type superclass, boolean avoidNestedTypeNameClashes) {
```

Comment:
Would we ever want to _not_ avoid nested type name clashes? Feels like this should just be the default behavior and we don't need the overloads.

Reply:
I think it's invasive enough of a change that it's worth an off switch. Otherwise I think it merits an RC first so some major consumers could give it a spin

Reply:
Makes sense. Then I think the default should false to ensure that the behavior of `superclass(Type)` is unchanged.

Reply:
FWIW I'm upgrading Google's internal copy of JavaPoet at the moment and running into a number of problems because of this. I think the default _should_ have been false, but I guess it's too late now.